client-cm/src/main/java/com/sequenceiq/cloudbreak/cm/client/tracing/CmOkHttpTracingInterceptor.java

```diff
@@ -0,0 +1,61 @@
+package com.sequenceiq.cloudbreak.cm.client.tracing;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.cloudbreak.tracing.TracingUtil;
+import com.squareup.okhttp.Interceptor;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+import io.opentracing.References;
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+@Component
+public class CmOkHttpTracingInterceptor implements Interceptor {
+
+    private static final String CLOUDERA_MANAGER = "Cloudera Manager";
+
+    @Inject
+    private Tracer tracer;
+
+    @Inject
+    private StackTraceBasedCmApiNameExtractor stackTraceBasedCmApiNameExtractor;
+
+    @NotNull
+    @Override
+    public Response intercept(@NotNull Chain chain) throws IOException {
+        Optional<String> cmApiNameOptional = stackTraceBasedCmApiNameExtractor.getCmApiName(Thread.currentThread().getStackTrace());
+        Request request = chain.request();
+        String cmApiName = cmApiNameOptional.orElse(request.url().getPath());
+        Span span = tracer.buildSpan("CM - [" + cmApiName + "] ")
+                .addReference(References.FOLLOWS_FROM, tracer.activeSpan() != null ? tracer.activeSpan().context() : null)
+                .start();
+        span.setTag(TracingUtil.COMPONENT, CLOUDERA_MANAGER);
+        span.setTag(TracingUtil.URL, request.url().toString());
+        span.setTag(TracingUtil.HTTP_METHOD, request.method());
+        span.setTag(TracingUtil.HEADERS, Json.silent(chain.request().headers().toMultimap()).getValue());
```

Comment:
this could bre refactored into a separate method:
```
    private Span createSpan(Chain chain) {
        Request request = chain.request();
        Optional<String> cmApiNameOptional = stackTraceBasedCmApiNameExtractor.getCmApiName(Thread.currentThread().getStackTrace());
        String cmApiName = cmApiNameOptional.orElse(request.url().getPath());
        Span span = tracer.buildSpan("CM - [" + cmApiName + "] ")
                .addReference(References.FOLLOWS_FROM, tracer.activeSpan() != null ? tracer.activeSpan().context() : null)
                .start();
        span.setTag(TracingUtil.COMPONENT, CLOUDERA_MANAGER);
        span.setTag(TracingUtil.URL, request.url().toString());
        span.setTag(TracingUtil.HTTP_METHOD, request.method());
        span.setTag(TracingUtil.HEADERS, Json.silent(request.headers().toMultimap()).getValue());
        return span;
    }
```